### PR TITLE
SER-412-NT-improve-get-history

### PR DESCRIPTION
**Pull request by Nadin-a**
improved _getElectrumConfig method

### CODE REVIEW

File: src/electrum-client/define-electrum-client.ts [hunk 1]
```diff
@@ -38,10 +39,25 @@ async function getOptions(query: ConfigurationReqDTO | CoinTypeReqDTO): Promise<
...
+    for (let config of configs) {
+        const hostIsAvailable = await isPortReachable(config.port, {host: config.host});
+        if (hostIsAvailable) {
+            availableConfig = config;
+            break;
+        }
+    }
+
+    if (!availableConfig) {
+        throw 'No available configs';
```
Comment (Isahara86) [inline src/electrum-client/define-electrum-client.ts]: throw 'No available configs'; => throw Error('No available configs');

File: src/electrum-client/define-electrum-client.ts [hunk 2]
```diff
@@ -38,10 +39,25 @@ async function getOptions(query: ConfigurationReqDTO | CoinTypeReqDTO): Promise<
     }
 }
 
-function _getElectrumConfig(type: CoinType, netMode: Netmode): ElectrumConfig {
+async function _getElectrumConfig(type: CoinType, netMode: Netmode) {
```
Comment (Isahara86) [inline src/electrum-client/define-electrum-client.ts]: PLease add return type

File: tsconfig.json [hunk 3]
```diff
@@ -25,7 +25,7 @@
     /* Strict Type-Checking Options */
     "strict": true,                           /* Enable all strict type-checking options. */
     // "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
-    // "strictNullChecks": true,              /* Enable strict null checks. */
+    "strictNullChecks": false,              /* Enable strict null checks. */
```
Comment (Isahara86) [inline tsconfig.json]: doubtful param

File: src/electrum-client/define-electrum-client.ts [hunk 4]
```diff
@@ -17,6 +15,7 @@ async function defineElectrumClient(req: any, res: Response) {
         const protocol = defaultOptions.connectionType;
         const version = defaultOptions.version;
 
+        console.log(defaultOptions);
```
Comment (Isahara86) [inline src/electrum-client/define-electrum-client.ts]: remove console

File: src/electrum-client/electrum-config.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        throw Error('No available configs');
+    }
+    return availableConfig;
+}
+
+async function getElectrumServers(type: CoinType, connectionType: ProtocolTypeEnum): Promise<ElectrumConfig[]> {
+    if (process.env.NODE_ENV === 'test') { return; }
+    if(!servers_json){ return []; }
+
+    let full_config_servers: RequestedServer[] = JSON.parse(servers_json).filter((server: any) => Object.keys(CoinType).includes(server.currency) &&
```
Comment (Isahara86) [inline src/electrum-client/electrum-config.ts]: Awful solution for performance. Please improve this

Comment (Nadin-a) [reply src/electrum-client/electrum-config.ts]: Changes:
- replaced filter to find;
- removed unnecessary conditions;
- refactored whole code;

**Pull request merged**